cql3/expr/expression.hh

```diff
@@ -183,14 +191,53 @@ struct function_call {
 
 struct cast {
     nested_expression arg;
-    cql3_type type;
+    std::variant<cql3_type, shared_ptr<cql3_type::raw>> type;
 };
 
 struct field_selection {
     nested_expression structure;
     shared_ptr<column_identifier_raw> field;
 };
 
+struct null {
+};
+
+struct bind_variable {
+    struct scalar {};
+    struct scalar_in {};
+    struct tuple {};
+    struct tuple_in {};
+    // FIXME: infer shape from expression rather than from grammar
+    std::variant<scalar, scalar_in, tuple, tuple_in> shape;
+    int32_t bind_index;
+};
+
+// A constant which does not yet have a date type. It is partially typed
+// (we know if it's floating or int) but not sized.
+struct untyped_constant {
+    enum type_class { integer, floating_point, string, boolean, duration, uuid, hex };
+    type_class partial_type;
+    sstring raw_text;
+};
+
+// Constructs a tuple from elements
```

Comment:
Not sure I understand the blurb.  The whole point of POD structs is not to act ("constructs") but to represent.  I'd like to know what `tuple_constructor` represents.

Reply:
In the CQL world, this constructs a tuple. In the C++ world it's just a POD. I'll clarify the comment.

Reply:
Updated both the comment and the commit changelog.